trestlebot/cli/commands/create.py

```diff
@@ -0,0 +1,234 @@
+        trestle_root=repo_path,
+    )
+    authored_comp.create_new_default(
+        profile_name=profile_name,
+        compdef_name=compdef_name,
+        comp_title=component_title,
+        comp_description=component_description,
+        comp_type=component_definition_type,
+        filter_by_profile=filter_by_profile,
+    )
+    logger.info(f"Component definition name is: {component_title}.")
+
+    transformer: ToRulesYAMLTransformer = ToRulesYAMLTransformer()
+
+    model_filter: ModelFilter = ModelFilter(
+        [], [profile_name, component_title, f"{const.RULE_PREFIX}*"]
+    )
+
+    rule_transform_task: RuleTransformTask = RuleTransformTask(
+        working_dir=repo_path,
+        rules_view_dir=const.RULES_VIEW_DIR,
+        rule_transformer=transformer,
+        model_filter=model_filter,
+    )
+    logger.info(
+        f"Profile to filter controls in the component files is: {filter_by_profile}."
+    )
+    logger.debug(
+        f"Oscal profile in use with the component definition is: {profile_name}."
+    )
+    logger.debug(f"Component definition type is {component_definition_type}.")
+
+    pre_tasks.append(rule_transform_task)
+
+    regenerate_task: RegenerateTask = RegenerateTask(
+        authored_object=authored_comp,
+        markdown_dir=markdown_dir,
+        model_filter=model_filter,
+    )
+    pre_tasks.append(regenerate_task)
+
+    run(pre_tasks, kwargs)
+    logger.debug(f"You have successfully authored the the {compdef_name}.")
+
+
+@create_cmd.command("ssp")
+@click.pass_context
+@common_create_options
+@common_options
+@click.option(
+    "--ssp-name",
+    prompt="Enter name of SSP to create",
+    help="Name of SSP to create.",
+)
+@click.option(
+    "--leveraged-ssp",
+    required=False,
+    type=str,
+    help="Provider SSP to leverage for the new SSP.",
+)
+@click.option(
+    "--ssp-index-file",
+    required=False,
+    type=str,
+    default="ssp-index.json",
+    help="Optionally set the path to the SSP index file.",
+)
+@click.option(
+    "--yaml-header-path",
+    required=False,
+    type=str,
+    default="ssp-index.json",
+    help="Optionally set a path to a YAML file for custom SSP Markdown YAML headers.",
+)
+@click.option(
+    "--version",
+    required=False,
+    type=str,
+    help="Optionally set the version of the SSP.",
+)
+@click.option(
+    "--compdefs",
+    required=False,
+    type=str,
+    help="Comma separated list of component definitions.",
+)
+@handle_exceptions
+def ssp_cmd(
+    ctx: click.Context,
+    **kwargs: Any,
+) -> None:
+    """
+    SSP Authoring command
+    """
+
+    profile_name = kwargs["profile_name"]
+    ssp_name = kwargs["ssp_name"]
+    leveraged_ssp = kwargs["leveraged_ssp"]
+    ssp_index_file = kwargs["ssp_index_file"]
+    yaml_header_path = kwargs["yaml_header_path"]
+    repo_path = kwargs["repo_path"]
+    markdown_dir = kwargs["markdown_dir"]
+    compdefs = kwargs["compdefs"]
+    version = kwargs["version"]
```

Comment:
For optional options, here may raise 'KeyError' exception.

Reply:
Yes good catch, optional args should use the `.get(<attr>, <default>)` method.  Also, if the variable is only used once it may not be worth instantiating a variable and instead just using `kwargs[arg]` in the one place it is needed.  Ex. leveraged_ssp.